src/main/java/ch/wisv/areafiftylan/web/banner/controller/BannerController.java

```diff
@@ -0,0 +1,112 @@
+package ch.wisv.areafiftylan.web.banner.controller;
+
+import ch.wisv.areafiftylan.exception.BannerNotFoundException;
+import ch.wisv.areafiftylan.web.banner.model.Banner;
+import ch.wisv.areafiftylan.web.banner.service.BannerService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+
+import static ch.wisv.areafiftylan.utils.ResponseEntityBuilder.createResponseEntity;
+
+@Controller
+@RequestMapping("/web/banners")
+public class BannerController {
+
+    private BannerService bannerService;
+
+    public BannerController(BannerService bannerService) {
+        this.bannerService = bannerService;
+    }
+
+    /**
+     * Get all banners in the database.
+     *
+     * @return all banners
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping
+    public ResponseEntity<?> getBanners() {
+        Collection<Banner> banners = bannerService.getAllBanners();
+        return new ResponseEntity<>(banners, HttpStatus.OK);
+    }
+
+    /**
+     * Returns a banner where the current date (date when the request was made)
+     * lies within the range of its start- and end-date.
+     *
+     * @return a banner
+     */
+    @GetMapping("/current")
+    public ResponseEntity<?> getCurrentBanner() {
+        Banner currentBanner = bannerService.getCurrentbanner();
+        return new ResponseEntity<>(currentBanner, HttpStatus.OK);
```

Comment:
We can use the createResponseEntity() helper method like you did below.

Reply:
Done